### PR TITLE
211 convert admin side api calls to graphql

### LINKED ISSUE #211: Move Admin side api calls to graphql

### PR DESCRIPTION
**Pull request by sudeeptarlekar**
This fixes #211

@breezyboa _a Please review

### CODE REVIEW

File: app/javascript/Admin/Request/UrlRequest.elm [hunk 1]
```diff
@@ -63,19 +47,3 @@ createUrl nodeEnv apiKey body =
             JsonDecoder.field "_id" JsonDecoder.string
     in
         httpRequest requestData decoder
-
-
-createUrlMutation : GQLBuilder.Document GQLBuilder.Mutation UrlData CreateUrlInput
-createUrlMutation =
```
Comment (breezyboa) [inline app/javascript/Admin/Request/UrlRequest.elm]: Why was this mutation removed?

File: app/javascript/Admin/Request/CategoryRequest.elm [hunk 2]
```diff
@@ -1,61 +1,22 @@
...
+requestCategories =
+    Reader.Reader
+        (\( env, apiKey ) ->
+            GQLClient.sendQuery (graphqlUrl env) <|
+                GQLBuilder.request {} categoriesQuery
         )
-
-
-createCategoryMutation : GQLBuilder.Document GQLBuilder.Mutation Category CreateCategoryInputs
-createCategoryMutation =
```
Comment (breezyboa) [inline app/javascript/Admin/Request/CategoryRequest.elm]: Why was this removed?

File: app/javascript/Admin/Page/Category/List.elm [hunk 3]
```diff
@@ -6,32 +6,35 @@ import Html.Attributes exposing (..)
...
-    { categoryList = { categories = [] }
+    { categories = []
     , errors = Nothing
     }
 
 
-init : ( Model, Cmd Msg )
-init =
+init : NodeEnv -> ApiKey -> ( Model, Cmd Msg )
+init env key =
```
Comment (breezyboa) [inline app/javascript/Admin/Page/Category/List.elm]: You can look at this in a separate issue
But by using a Reader we do not have to pass down these arguments (NodeEnv and ApiKey)
Rather we can handle it at the very end i.e in the main module.
If we follow this in all pages we have a uniform pattern
and the init function signature would just be
```
init : (Model, Reader ( NodeEnv, ApiKey ) (Task.Task GQLClient.Error Article) )
```

suppose init needs anything extra we can just pass those down 
```
init : ArticleId -> Reader ( NodeEnv, ApiKey ) (Task.Task GQLClient.Error Article)
```

**Comment by breezyboa** [comment]
@sudeeptarlekar _a
can you check why build is failing

File: app/javascript/Admin/Request/UrlRequest.elm (hunk 1 above)
Comment (sudeeptarlekar) [reply app/javascript/Admin/Request/UrlRequest.elm]: Its not been removed basically its been moved to correct file. `UrlData.elm`

File: app/javascript/Admin/Request/CategoryRequest.elm (hunk 2 above)
Comment (sudeeptarlekar) [reply app/javascript/Admin/Request/CategoryRequest.elm]: this is actually moved to correct folder `CategoryData.elm`

File: app/javascript/Admin/Page/Category/List.elm (hunk 3 above)
Comment (sudeeptarlekar) [reply app/javascript/Admin/Page/Category/List.elm]: created issue #227 for same

**Comment by sudeeptarlekar** [comment]
@breezyboa _a builds are passing now, pls take a look
